src/mpyl/steps/__init__.py

```diff
@@ -105,13 +105,13 @@ class Step(metaclass=IPluginRegistry):
     """Is set to something other than `ArtifactType.NONE` if this step depends on an artifact produced by the execution
     of an earlier step. For example: a step in the `Deploy` stage, may need to deploy a docker image that was produced
     in the `Build` stage."""
-    before: Optional[Step]
-    after: Optional[Step]
+    before: Optional[list[Step]]
+    after: Optional[list[Step]]
```

Comment:
Wondering if an optional list makes sense.. Is there a difference between a `None` and an empty list? If there're no steps before or after, should be an empty list I think

Reply:
Discussed with Sam, will revert this and add a `Scan Stage` instead.